rfd/0174-kube-rbac-bootstrap.md

```diff
@@ -0,0 +1,346 @@
+
+If `kubernetes_permissions` field is set in the `allow` part of the Teleport role, fields `kubernetes_resources`, `kubernetes_users` and `kubernetes_groups` 
+should not be set, we will return an error if user will try to create a role mixing those fields. In the future we might relax this rule.
+
+The field `kubernetes_permissions` will be forbidden from appearing in the `deny` part of the role definition since we are trying to keep its meaning close to
+native Kubernetes RBAC, which can only allow access. Although the field `kubernetes_resource` could appear in the `deny` part of the role, 
+even if `kubernetes_permissions` is set in the `allow` part, giving user more flexibility using Teleport-level permissions check.
+
+Pattern matching for resource names that is supported by the `kubernetes_resources` field will not be supported (same as in native Kubernetes RBAC)
+by the `kubernetes_permissions` in the
+first implementation, but we will be able to add it later. Not allowing pattern matching, though limiting functionality compared to the `kubernetes_resources` field,
+will allow us to translate permissions defined in Teleport 1-to-1 into Kubernetes RBAC definitions. Since this feature is targeted primarily for the new users of 
+Teleport and/or users with less complicated RBAC setups, this should not be a critical issue. If we add pattern matching later, it will need to rely on 
+additional checks at the Teleport level - the way it currently works for the `kubernetes_resources`.
+
+Kubernetes RBAC will be auto-provisioned using the KubeProvisions functionality described in the next section of this RFD. For the best UX any change to
+a role that contains `kubernetes_permissions` will trigger auto-provisioning immediately to reduce latency between the moments when role was created 
+or changed and when clients can use Kubernetes cluster with the updated RBAC.
+
+Introducing `kubernetes_permissions` field will make enrolling into Teleport and starting using Kubernetes clusters
+a much simpler process, since user will be able to setup RBAC completely in Teleport if they want. We will also be able
+to amend our documentation to make our guides simpler, which will also help users to start with Teleport more easily. 
+
+#### UX
+
+Let's look at an example user story. Alice is a system administrator and she explores Teleport capabilities.
+She doesn't know much about Teleport yet, except for basic foundation (agents, roles). She just enrolled her first Kubernetes cluster into Teleport.
+This cluster contains the staging environment and she wants to give read-only pod access to the developers in an app namespace.
+She goes to the UI and creates a new role, putting this content into it:
+
+```yaml
+kind: role
+metadata:
+  name: staging-kube-access
+version: v7
+spec:
+  allow:
+    kubernetes_labels:
+      'env': 'staging'
+    kubernetes_permissions:
+      - namespaces: ["main-company-app"]
```

Comment:
will we support namespace matching?

Reply:
You mean something like `- namespaces: ["env-*"]`? I think we can, but not sure if we should include it in the first version, since we don't include pattern matching in the resource names, so maybe just keep it simple at start 🤔